src/database.py

```diff
@@ -80,8 +80,13 @@ def write_billing_event_to_database(event, db_connection):
         getLogger('event-recorder').warning('Failed to store a billing event [Event ID {0}] due to key error'.format(event.event_id))
         raise keyError
     except IntegrityError as integrityError:
-        getLogger('event-recorder').warning('Failed to store a billing event [Event ID {0}] due to integrity error'.format(event.event_id))
-        raise integrityError
+        if integrityError.pgcode == UNIQUE_VIOLATION:
+            # The event has already been recorded - don't throw an exception (no need to retry this message), just
```

Comment:
Does this happen? If so, why?

Reply:
Under normal operation, no, it doesn't happen ... however, it has the potential to occur if we have to replay events for any reason, which is why added the same catch for this in the `write_audit_event_to_database()` method. Now that we have a primary key on `billing_events`, it seems sensible to have the same catch for in this method.